src/textual/cli/_run.py

```diff
@@ -0,0 +1,134 @@
+"""
+
+Functions to run Textual apps with an updated environment.
+
+Note that these methods will execute apps in a new process, and abandon the current process.
+This means that (if they succeed) they will never return.
+
+"""
+
+from __future__ import annotations
+
+import os
+import platform
+import shlex
+import sys
+from string import Template
+from typing import NoReturn
+
+WINDOWS = platform.system() == "Windows"
+"""True if we're running on Windows."""
+
+EXEC_SCRIPT = Template(
+    """\
+from textual.app import App
+from $MODULE import $APP as app;
+if isinstance(app, App):
+    # If we imported an app, run it
+    app.run()
+else:
+    # Otherwise it is assumed to be a class
+    app().run()
+"""
+)
+"""A template script to import and run an app."""
+
+
+def run_app(command_args: str, environment: dict[str, str] | None = None) -> None:
+    """Run a textual app.
+
+    Note:
+        The current process is abandoned.
+
+    Args:
+        import_name: Path or import:App of Textual app.
+        args: Arguments to pass to the Textual app.
+        environment: Environment variables, or None to use current process.
+    """
+    import_name, *args = shlex.split(command_args, posix=not WINDOWS)
+    if environment is None:
+        app_environment = dict(os.environ)
+    else:
+        app_environment = environment
+
+    if _is_python_path(import_name):
+        # If it is a Python path we can exec it now
+        exec_python([import_name, *args], app_environment)
+
+    else:
+        # Otherwise it is assumed to be a Python import
+        try:
+            exec_import(import_name, args, app_environment)
+        except SyntaxError:
+            print(f"Unable to import Textual app from {import_name!r}")
+
+
+def _is_python_path(path: str) -> bool:
+    """Does the given file look like it's run with Python?
+
+    Args:
+        path: The path to check.
+
+    Returns:
+        True if the path references Python code, False it it doesn't.
+    """
+    if path.endswith(".py"):
+        return True
+    try:
+        with open(path, "r") as source:
+            first_line = source.readline()
+    except IOError:
+        return False
+    return first_line.startswith("#!") and "py" in first_line
```

Comment:
I'll be honest and say I'm struggling to think of one off the top of my head right now, although of course "#!" can point to *anything*,  but I feel that looking for just `py` is a little too permissive here and could result in a false positive.

Perhaps consider being just a wee bit more guarded here?

Reply:
Will go with this for now, and handle any clashes if they occur.